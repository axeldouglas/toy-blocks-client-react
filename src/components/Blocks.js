import React from "react";
import PropTypes from "prop-types";
import {
  Typography,
  makeStyles,
  Box,
  CircularProgress,
} from "@material-ui/core";
import colors from "../constants/colors";
import Block from "./Block";

const Blocks = ({ node }) => {
  const classes = useStyles();

  if (
    !node.blocks.length &&
    (node.loading === true || node.loadingBlocks === true)
  ) {
    return (
      <Box className={classes.spinnerWrapper}>
        <CircularProgress color="inherit" />
      </Box>
    );
  }

  if (!node.blocks.length) {
    return (
      <Typography className={classes.notFound}>No results found</Typography>
    );
  }

  return (
    <Box className={classes.blockWrapper}>
      {node.blocks.map((block) => (
        <Block key={block.id} block={block} />
      ))}
    </Box>
  );
};

const useStyles = makeStyles((theme) => ({
  blockWrapper: {
    display: "flex",
    flexDirection: "column",
    width: "100%",
  },
  spinnerWrapper: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
    color: "rgba(0, 0, 0, 0.12)",
  },
  notFound: {
    width: "100%",
    textAlign: "center",
    padding: "10px 0 14px",
    fontSize: theme.typography.pxToRem(14),
    color: colors.text,
    opacity: 0.54,
  },
}));

Blocks.propTypes = {
  node: PropTypes.shape({
    url: PropTypes.string,
    online: PropTypes.bool,
    name: PropTypes.string,
    loading: PropTypes.bool,
    loadingBlocks: PropTypes.bool,
    blocks: PropTypes.array,
  }).isRequired,
};

export default Blocks;
