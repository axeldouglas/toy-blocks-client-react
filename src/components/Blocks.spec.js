import { mount } from "enzyme";
import React from "react";
import { create } from "react-test-renderer";
import { CircularProgress } from "@material-ui/core";
import Blocks from "./Blocks";

describe("<Blocks />", () => {
  const blocks = [
    {
      id: "5",
      type: "blocks",
      attributes: {
        index: 1,
        timestamp: 1530679678,
        data: "The Human",
        "previous-hash": "IoVe8=",
        hash: "oHkxOJW=",
      },
    },
  ];

  const node = {
    url: "https://localhost:3003",
    online: true,
    name: "Node 1",
    loading: false,
    loadingBlocks: false,
    blocks,
  };

  const nodeLoading = {
    url: "https://localhost:3003",
    online: true,
    name: "Node 1",
    loading: false,
    loadingBlocks: true,
    blocks: [],
  };

  const nodeEmpty = {
    url: "https://localhost:3003",
    online: true,
    name: "Node 1",
    loading: false,
    loadingBlocks: false,
    blocks: [],
  };

  it("should contain <CircularProgress />", () => {
    const wrapper = mount(<Blocks node={nodeLoading} />);

    expect(wrapper.find(CircularProgress).length).toEqual(1);
  });

  it("should match empty data snapshot", () => {
    const component = create(<Blocks node={nodeEmpty} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    const component = create(<Blocks node={node} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
