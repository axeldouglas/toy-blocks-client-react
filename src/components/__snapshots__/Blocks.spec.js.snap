// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blocks /> should match empty data snapshot 1`] = `
<p
  className="MuiTypography-root makeStyles-notFound-3 MuiTypography-body1"
>
  No results found
</p>
`;

exports[`<Blocks /> should match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-5 makeStyles-blockWrapper-1"
>
  <div
    className="MuiBox-root MuiBox-root-9 makeStyles-blockContainer-6"
  >
    <p
      className="MuiTypography-root makeStyles-blockTitle-7 MuiTypography-body1"
    >
      005
    </p>
    <p
      className="MuiTypography-root makeStyles-blockDescription-8 MuiTypography-body1"
    >
      The Human
    </p>
  </div>
</div>
`;
